pythonforandroid/recipes/libsodium/__init__.py

```diff
@@ -3,15 +3,25 @@
 from pythonforandroid.logger import shprint
 from multiprocessing import cpu_count
 import sh
+from packaging import version as packaging_version
 
 
 class LibsodiumRecipe(Recipe):
     version = '1.0.16'
-    url = 'https://github.com/jedisct1/libsodium/releases/download/{version}/libsodium-{version}.tar.gz'
+    url = 'https://github.com/jedisct1/libsodium/releases/download/{}/libsodium-{}.tar.gz'
     depends = []
     patches = ['size_max_fix.patch']
     built_libraries = {'libsodium.so': 'src/libsodium/.libs'}
 
+    @property
+    def versioned_url(self):
+        asked_version = packaging_version.parse(self.version)
+        recipe_version = packaging_version.parse(self._version)
+        if asked_version > recipe_version:
+            return self._url.format(self.version + '-RELEASE', self.version)
+        else:
+            return self._url.format(self.version, self.version)
```

Comment:
Shouldn't we hardcode the value rather than using `recipe_version`?
```
        if asked_version > packaging_version.parse("1.0.16"):
            return self._url.format(self.version + '-RELEASE', self.version)
        else:
            return self._url.format(self.version, self.version)
```
Otherwise if someone bumps the default recipe version it wouldn't work

Reply:
Yes, you are correct.